pandas/core/dtypes/common.py

```diff
@@ -1114,30 +1112,27 @@ def needs_i8_conversion(arr_or_dtype) -> bool:
     >>> needs_i8_conversion(np.int64)
     False
     >>> needs_i8_conversion(np.datetime64)
+    False
+    >>> needs_i8_conversion(np.dtype(np.datetime64))
     True
     >>> needs_i8_conversion(np.array(['a', 'b']))
     False
     >>> needs_i8_conversion(pd.Series([1, 2]))
     False
     >>> needs_i8_conversion(pd.Series([], dtype="timedelta64[ns]"))
-    True
+    False
```

Comment:
I guess this didn't meaningfully affect existing usage, or was this always supposed to be false?

Reply:
The existing implementation, like the is_foo_dtype functions, will unwrap array-likes.  We've changed most of our usages to not need that.  This just enforces that (which is why we don't really get a perf bump)